src/Share/Share.router.ts

```diff
@@ -0,0 +1,63 @@
+import { Response } from 'express'
+import { env } from 'decentraland-commons'
+import { format } from 'util'
+
+import { Router } from '../common/Router'
+
+import { Project } from '../Project/Project.model'
+import { withSocialUserAgentDetector, SocialRequest } from '../middleware/share'
+import { Params } from './Share.types'
+import { Pool, PoolAttributes } from '../Pool'
+import { ProjectAttributes } from '../Project'
+import template from './template'
+
+const BUILDER_URL = env.get('BUILDER_URL', '')
+
+export class ShareRouter extends Router {
+  mount() {
+    /**
+     * Redirect to scene
+     */
+    this.router.get(
+      '/share/:type(scene|pool)/:id',
+      withSocialUserAgentDetector,
+      this.redirectToBuilder
+    )
+  }
+
+  private redirectToBuilder = async (req: SocialRequest, res: Response) => {
+    const { type, id } = req.params as Params
+    const targetPath = type === 'scene' ? `/view/${id}` : `/view/${type}/${id}`
+    const url = BUILDER_URL + targetPath
+    if (!req.socialAgent) {
+      return res.redirect(301, url)
+    }
+
+    const p = await this.findPoolOrProject(id, type)
```

Comment:
Same as below, maybe naming it `p` is shooting ourselves in the foot? What if tomorrow we add a new element that starts with a different letter.
Do we maybe have a name that encompasses both project and pools?